cmd/ci-operator-checkconfig/main.go

```diff
@@ -18,11 +22,47 @@ func main() {
 		os.Exit(1)
 	}
 
+	seen := map[api.ImageStreamTagReference][]*config.Info{}
 	if err := config.OperateOnCIOperatorConfigDir(configDir, func(configuration *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
 		// validation is implicit, so we don't need to do anything
+		// but record the images we saw for future validation
+		if configuration.PromotionConfiguration == nil {
```

Comment:
I think we want to keep this functionality, so it would be great to have a basic test for it :)

Reply:
As a unit test for the function passed to `OperateOn..`?

Reply:
Well, then we only test half of it. Wouldn't it make sense to factor out the whole of it so we can actually test it?

Reply:
Personally I would be fine with unit-testing the function passed to `OperateOn`.

Reply:
Agreed with @alvaroaleman, factoring out seems like the best option here.